Runtime/Scripts/MeshSyncServer.Properties.cs

```diff
@@ -0,0 +1,300 @@
+        {
+            changedMeshes.Add(mesh);
+        }
+
+        static GetFlags sendMeshSettings = new GetFlags(new GetFlags.GetFlagsSetting[] {
+                                   GetFlags.GetFlagsSetting.Transform,
+                                   GetFlags.GetFlagsSetting.Points,
+                                   GetFlags.GetFlagsSetting.Indices,
+                                   GetFlags.GetFlagsSetting.MaterialIDS,
+            });
+
+        void SendMesh(string path, EntityRecord entity)
+        {
+            Debug.Assert(entity.dataType == EntityType.Mesh);
+
+            var objRenderer = entity.meshRenderer;
+
+            MeshData dst = MeshData.Create();
+
+            var mesh = entity.meshFilter.sharedMesh;
+            var mr = entity.meshRenderer;
+
+            CaptureMesh(ref dst, mesh, null,
+                sendMeshSettings,
+                mr.sharedMaterials);
+            
+            TransformData dstTrans = dst.transform;
+            Transform rendererTransform = objRenderer.transform;
+            dstTrans.hostID = GetObjectlID(objRenderer.gameObject);
+            dstTrans.position = rendererTransform.localPosition;
+            dstTrans.rotation = rendererTransform.localRotation;
+            dstTrans.scale = rendererTransform.localScale;
+            dst.local2world = rendererTransform.localToWorldMatrix;
+            dst.world2local = rendererTransform.worldToLocalMatrix;
+
+            dstTrans.path = path;
+
+            m_server.SendMesh(dst);
+        }
+#endif
+
+#if AT_USE_SPLINES
+
+        void SendCurve(string path, EntityRecord entity)
+        {
+            Debug.Assert(entity.dataType == EntityType.Curve);
+            
+            for (int splineIdx = 0; splineIdx < entity.splineContainer.Splines.Count; splineIdx++)
+            {
+                var spline = entity.splineContainer.Splines[splineIdx];
+
+                var pointCount = spline.Count;
+                if (pointCount == 0)
+                {
+                    continue;
+                }
+
+                var cos = new float3[pointCount];
+                var handlesLeft = new float3[pointCount];
+                var handlesRight = new float3[pointCount];
+
+                for (int pointIdx = 0; pointIdx < pointCount; pointIdx++)
+                {
+                    var knot = spline[pointIdx];
+
+                    var co = knot.Position;
+
+                    cos[pointIdx] = co;
+
+                    handlesLeft[pointIdx] = math.rotate(knot.Rotation, knot.TangentIn) + co;
+                    handlesRight[pointIdx] = math.rotate(knot.Rotation, knot.TangentOut) + co;
+                }
+
+                m_server.SendCurve(path, splineIdx, pointCount, spline.Closed, cos, handlesLeft, handlesRight);
+            }
+        }
+
+        protected override void SplineChanged(Spline spline, int arg2, SplineModification arg3)
+        {
+            // TODO: Get the path here:
```

Comment:
Yes because the current API only gives us the unity spline object that has changed and we have no good way to map that back to the spline that changed so we send all splines back. We could optimise that to only send the changed spline back if we can get the object path of it somehow. I've asked the teamm to add something like a custom field to attach user data to these data structures.